src/app/components/Search/SearchForm.jsx

```diff
@@ -1,56 +1,132 @@
 
-    this.state = Store.getState();
+    this.state = props;
 
-    this.handleChange = this.handleChange.bind(this);
+    this.onFilterChange = this.onFilterChange.bind(this);
+    this.onQueryChange = this.onQueryChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+    this.submitSearchRequest = this.submitSearchRequest.bind(this);
   }
 
-  handleChange(event) {
-    console.log(event);
+  onFilterChange(event) {
+    const filterSelected = event.target.value;
+    this.setState({ filter: filterSelected });
+  }
+
+  onQueryChange(event) {
+    this.setState({ query: event.target.value });
   }
 
   handleSubmit(event) {
-    console.log(event);
+    if (event && event.charCode === 13) {
+      this.submitSearchRequest(event);
+    }
+  }
+
+  submitSearchRequest(event) {
+    event.preventDefault();
+    if (!this.state.filter) {
+      this.setState({ filter: 'q' });
+    }
+
+    this.state.search(this.state.query, this.state.filter);
   }
 
   render() {
+    const { query, filter, allowedFilters } = this.state;
+
     return (
       <div>
-        <form onSubmit={this.handleOnSubmit}>
-          <div className="sfr-search-form">
-            <fieldset className="nypl-omnisearch">
-              <span className="nypl-omni-fields">
-                <label htmlFor="search-by-field">Search in</label>
-                <select id="search-by-field">
-                  <option value="q" defaultValue>Keyword</option>
-                  <option value="filters[title]">Title</option>
-                  <option value="filters[author]">Author</option>
-                </select>
-              </span>
-              <input
-                type="text"
-                aria-labelledby="nypl-omni-button"
-                value={this.state.value}
-                placeholder="Keyword, title, or author"
-                onChange={this.handleChange}
-              />
-              <input
-                type="submit"
-                id="nypl-omni-button"
-                value="Search"
-              />
-            </fieldset>
+        <form className="nypl-omnisearch-form" onSubmit={this.handleSubmit} onKeyPress={this.handleSubmit}>
+          <div className="ebook-search-form">
+            <div className="nypl-omnisearch">
+              <div className="nypl-text-field">
+                <span className="nypl-omni-fields">
+                  <label htmlFor="search-by-field">Search in</label>
+                  <select
+                    id="search-by-field"
+                    onChange={this.onFilterChange}
+                    value={filter}
+                  >
+                    <option value={allowedFilters.kw} defaultValue>
+                      Keyword
+                    </option>
+                    <option value={allowedFilters.ti}>
+                      Title
+                    </option>
+                    <option value={allowedFilters.au}>
+                      Author
+                    </option>
```

Comment:
Sorry to get hung up on this but I think these "filters" are better termed "queries/search queries" since that's the block they'll end up getting passed to in Elastic. Filters will be a separate block in the API. Just trying to anticipate variables that might end up being confusing.